### PR TITLE
Step List

### PR DESCRIPTION
**Pull request by martypowell**
Addresses the step-list component

- Adds logic for step list (will identify based on markup) on page load
- Adds integration tests to confirm the functionality
- Markdown file on instructions on how to use the step list

### CODE REVIEW

File: src/components/StepList.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                <p>Step 3 details.</p>
+            </div>
+        </li>
+    </ol>
+</div>
+```
+
+The only pieces of markup that should be modified are:
+
+**Button**
```
Comment (martypowell) [inline src/components/StepList.md]: this needs to be updated aria-labelledby also comes into the picture here.

File: src/components/StepList.js [hunk 2]
```diff
@@ -0,0 +1,392 @@
+import { GetFirstElementOrDefault } from "../utilities/dom.utils";
+
+// Production steps of ECMA-262, Edition 6, 22.1.2.1
+if (!Array.from) {
```
Comment (martypowell) [inline src/components/StepList.js]: this needs to be moved out of here, into a polyfill.

File: dist/demos/components/StepList.css [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  text-decoration: none;
+}
+
+.dg_step-list__show-all-btn {
+  margin-bottom: 15px;
+}
+
+.dg_step-list__details {
+  margin-top: 15px;
+}
```
Comment (tmccoy529) [inline dist/demos/components/StepList.css]: Can some of these values be converted to variables?

Comment (tmccoy529) [reply dist/demos/components/StepList.css]: Nevermind. I forgot this was the javascript project

Comment (martypowell) [reply dist/demos/components/StepList.css]: yep, but not a problem appreciate the comment.

**Pull request merged**